micro_http/src/server.rs

```diff
@@ -0,0 +1,551 @@
+                    Err(ConnectionError::ParseError(_error)) => {
+                        // An error occurred while parsing the read bytes.
+                        // Check if there are any valid parsed requests in the queue.
+                        while let Some(request) =
+                            file_streamed_connection.connection.pop_parsed_request()
+                        {
+                            // Add all valid requests to `parsed_requests` along with their token.
+                            parsed_requests.push((token, request));
+                        }
+
+                        // Send an error response for the request that gave us the error.
+                        let error_response = Response::new(Version::Http11, StatusCode::BadRequest);
+                        file_streamed_connection
+                            .connection
+                            .enqueue_response(error_response);
+                    }
+                    Err(ConnectionError::InvalidWrite) => {
+                        unreachable!();
+                    }
+                    Ok(()) => {
+                        while let Some(request) =
+                            file_streamed_connection.connection.pop_parsed_request()
+                        {
+                            // Add all valid requests to `parsed_requests` along with their token.
+                            parsed_requests.push((token, request));
+                        }
+                    }
+                }
+
+                // Check if we need to write any bytes for this connection.
+                if file_streamed_connection.connection.is_outgoing() {
+                    stream_direction = StreamDirection::Outgoing;
+                }
+            }
+
+            if event & epoll::Events::EPOLLOUT.bits() != 0 {
+                // The stream is available for writing.
+                match file_streamed_connection.connection.try_write() {
+                    Err(ConnectionError::ConnectionClosed)
+                    | Err(ConnectionError::StreamError(_)) => {
+                        // Writing to the stream failed so it will be removed.
+                        streams_to_be_removed.push(token);
+                    }
+                    Err(ConnectionError::InvalidWrite) => {
+                        // A `try_write` call was performed on a connection that has nothing
+                        // to write.
+                        panic!("Invalid write attempt.");
+                    }
+                    _ => {
+                        // Check if we still have bytes to write for this connection.
+                        if !file_streamed_connection.connection.is_outgoing() {
+                            stream_direction = StreamDirection::Incoming;
+                        }
+                    }
+                }
+            }
+        }
+
+        if let Some(file_streamed_connection) = self.connections.get(&token) {
```

Comment:
This can be moved out and depending on `conn.state()`:
```
match conn.state() {
    ConnState::awaiting_incoming => self.epoll_mod(fd, epoll::Events::EPOLLIN),
    ConnState::awaiting_outgoing => self.epoll_mod(fd, epoll::Events::EPOLLOUT),
    ConnState::closed => self.connections.remove(fd),
};
```